### PR TITLE
gossip: reduce verbosity of spammy message

### LINKED ISSUE #115637: Node shutdown: Remove spammy message from default logging
Per [this](https://cockroachlabs.slack.com/archives/C05VAJ5H3QS/p1701799779768419) thread, on node shutdown we generate the following log message a lot which makes the logs hard to follow in the normal case. We should consider lowering this to debug/trace level. 

`I231205 18:07:42.985117 20036 gossip/gossip.go:346 ⋮ [T1,Vsystem,n5] 37094  NodeDescriptor set to ‹node_id:5 address:<network_field:"tcp" address_field:"10.14.13.187:26258" > attrs:<> locality:<tiers:<key:"country" value:"us" > tiers:<key:"region" value:"us-west-2" > tiers:<key:"az" value:"us-west-2a" > tiers:<key:"pg" value:"us-west-2a1" > > ServerVersion:<major_val:1000023 minor_val:1 patch:0 internal:32 > build_tag:"v23.2.0-alpha.7-dev-0456c0d4e4dca39bde1b4bc146bf7158db3c0f30-dirty" started_at:1701745573418761051 cluster_name:"seanc-crdb23210tb" sql_address:<network_field:"tcp" address_field:"crdb23210tb-sql.seanc.dev:26257" > http_address:<network_field:"tcp" address_field:"10.14.13.187:8080" > ›`

Jira issue: CRDB-34136

Epic CRDB-34227

### PR DESCRIPTION
**Pull request by shralex**
This PR is lowering the verbosity of a Gossip-related log message.

Fixes: https://github.com/cockroachdb/cockroach/issues/115637

### CODE REVIEW

**Review by nvb — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/117191)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @shralex)

___
*[`pkg/gossip/gossip.go` line 346 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/117191#-NnFUp7k7SH84TD6iEWT:-NnFUp7k7SH84TD6iEWU:b6tnhf2) ([raw file](https://github.com/cockroachdb/cockroach/blob/145339d8ad067ea31f6d2b74536319fa37ea1250/pkg/gossip/gossip.go#L346)):*
> ```Go
> func (g *Gossip) SetNodeDescriptor(desc *roachpb.NodeDescriptor) error {
> 	ctx := g.AnnotateCtx(context.TODO())
> 	log.VInfof(ctx, 2, "NodeDescriptor set to %+v", desc)
> ```

Let's make this a verbosity of 1, like the rest of these messages.

**Review by shralex — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/117191)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @nvanbenschoten)

___
*[`pkg/gossip/gossip.go` line 346 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/117191#-NnFUp7k7SH84TD6iEWT:-NnF_GmeAaEAW5ugyGsL:bb6un1j) ([raw file](https://github.com/cockroachdb/cockroach/blob/145339d8ad067ea31f6d2b74536319fa37ea1250/pkg/gossip/gossip.go#L346)):*
<details><summary><i>Previously, nvanbenschoten (Nathan VanBenschoten) wrote…</i></summary><blockquote>

Let's make this a verbosity of 1, like the rest of these messages.
</blockquote></details>

Done
